haystack/preview/telemetry/telemetry.py

```diff
@@ -0,0 +1,197 @@
+        """
+        return {
+            "libraries.pytest": sys.modules["pytest"].__version__ if "pytest" in sys.modules.keys() else False,
+            "libraries.ipython": sys.modules["ipython"].__version__ if "ipython" in sys.modules.keys() else False,
+            "libraries.colab": sys.modules["google.colab"].__version__
+            if "google.colab" in sys.modules.keys()
+            else False,
+        }
+
+    def send_event(self, event_name: str, event_properties: Optional[Dict[str, Any]] = None):
+        """
+        Sends a telemetry event.
+
+        :param event_name: The name of the event to show in PostHog.
+        :param event_properties: Additional event metadata. These are merged with the
+            system metadata collected in __init__, so take care not to overwrite them.
+        """
+        event_properties = event_properties or {}
+        dynamic_specs = self.collect_dynamic_system_specs()
+        try:
+            posthog.capture(
+                distinct_id=self.user_id,
+                event=event_name,
+                # loads/dumps to sort the keys
+                properties=json.loads(
+                    json.dumps({**self.event_properties, **dynamic_specs, **event_properties}, sort_keys=True)
+                ),
+            )
+        except Exception as e:
+            logger.debug("Telemetry couldn't make a POST request to PostHog.", exc_info=e)
+
+
+def send_pipeline_run_event(pipeline: "Pipeline"):
+    """
+    Send a telemetry event for Pipeline.run(), if telemetry is enabled.
+    """
+    try:
+        if telemetry:
+            pipeline._telemetry_runs += 1
+            if pipeline._telemetry_runs in [1, 10, 100, 1000] or pipeline._telemetry_runs % 10_000 == 0:
+                pipeline_description = pipeline.to_dict()
+                components: Dict[str, List[Dict[str, Any]]] = {}
+                for component_name, component in pipeline_description["components"].items():
+                    if not component["type"] in components:
+                        components[component["type"]] = []
+                    components[component["type"]].append({"name": component_name})
+                telemetry.send_event("Pipeline run (2.x)", {"components": components, "runs": pipeline._telemetry_runs})
+    except Exception as e:
+        # Never let telemetry break things
+        logger.debug("There was an issue sending a 'Pipeline run (2.x)' telemetry event", exc_info=e)
+
+
+def tutorial_running(tutorial_id: str):
+    """
+    Send a telemetry event for a tutorial, if telemetry is enabled.
+    :param tutorial_id: identifier of the tutorial
+    """
+    try:
+        if telemetry:
+            telemetry.send_event("Tutorial", {"tutorial.id": tutorial_id})
+    except Exception as e:
+        # Never let telemetry break things
+        logger.debug("There was an issue sending a 'Tutorial' telemetry event", exc_info=e)
+
+
+if os.environ.get(HAYSTACK_TELEMETRY_ENABLED, "True") == "False":
```

Comment:
Can we check truthy values? Something like `telemetry_enabled = os.getenv("HAYSTACK_TELEMETRY_ENABLED", 'false').lower() in ('true', '1')`

Reply:
I copied it from old telemetry, so right now the behavior is identical. However I see no issue with this idea so I'm gonna do this :+1:

Reply:
Ah no there is an "issue": we want to count the variable unset as True, because most people don't set it. So it should be `os.getenv("HAYSTACK_TELEMETRY_ENABLED", 'true').lower() in ('true', '1')`. I'll change it to that, wdyt?

Reply:
`os.getenv("HAYSTACK_TELEMETRY_ENABLED", 'true').lower() in ('true', '1')` looks good 👍